Fix for Merge Duplicates Failing when no Determinations need Downgrading

MuchQuak opened the pull request:
# Issue #1451, #1482
# Summary
Add conditional check for if there are any determinations to downgrade before trying closes #1451

# Pull Request Checklist:

# Pre-Approval

- [x] There is a description section in the pull request that details what the proposed changes do. It can be very brief if need be, but it ought to exist.
- [x] Hotfixes should be branched off of the `master` branch and **squash and merged** back into the `master` branch.
- [x] Features and backlog bugs should be merged into the `Development` branch, **NOT** `master`
- [x] All new text is preferably internationalized (i.e., no end-user-visible text is hard-coded on the PHP pages), and the [spreadsheet tracking internationalizations](https://docs.google.com/spreadsheets/d/133fps9w2pUCEjUA6IGCcQotk7dn9KvepMXJ2IWUZsE8/edit?usp=sharing) has been updated either with a new row or with checkmarks to existing rows.
- [x] There are no linter errors
- [x] New features have responsive design (i.e., look aesthetically pleasing both full screen and with small or mobile screens)
- [x] [Symbiota coding standards](https://docs.google.com/document/d/1-FwCZP5Zu4f-bPwsKeVVsZErytALOJyA2szjbfSUjmc/edit?usp=sharing) have been followed
- [x] If any files have been reformatted (e.g., by an autoformatter), the reformat is its own, separate commit in the PR
- [x] Comment which GitHub issue(s), if any does this PR address
- [x] If this PR makes any changes that would require additional configuration of any Symbiota portals outside of the files tracked in this repository, make sure that those changes are detailed in [this document](https://docs.google.com/document/d/1T7xbXEf2bjjm-PMrlXpUBa69aTMAIROPXVqJqa2ow_I/edit?usp=sharing).

# Post-Approval

- [x] It is the code author's responsibility to merge their own pull request after it has been approved
- [x] If this PR represents a merge into the `Development` branch, remember to use the **squash & merge** option
- [x] If this PR represents a merge from the `Development` branch into the master branch, remember to use the **merge** option
- [x] If this PR represents a hotfix into the `master` branch, a subsequent PR from `master` into `Development` should be made **merge** option (i.e., no squash).
- [x] If the dev team has agreed that this PR represents the last PR going into the `Development` branch before a tagged release (i.e., before an imminent merge into the master branch), make sure to notify the team and [lock the `Development` branch](https://github.com/BioKIC/Symbiota/settings/branches) to prevent accidental merges while QA takes place. Follow the release protocol [here](https://github.com/BioKIC/Symbiota/blob/master/docs/release-protocol.md).
- [x] Don't forget to delete your feature branch upon merge. Ignore this step as required.

Thanks for contributing and keeping it clean!

MuchQuak committed:
add conditional check for if there are any determinations to downgrade before trying closes #1451

themerekat commented:
We have one more issue...
if the record being merged into (let's call this the "sink") has a determination, but the record with data that is being brought into the sink (let's call this the "source") does not have a determination, but it _does_ have a value in scientificname, we get a jumbled result.

This determination was in the "sink".
![image](https://github.com/BioKIC/Symbiota/assets/16688784/1755f4ab-76c0-464a-8b00-eb9ffbaeb700)

Then we wanted to merge it with the source that looked like this in the Occurrence Editor:
![image](https://github.com/BioKIC/Symbiota/assets/16688784/5e199e01-3ccc-4be5-964d-7ae3cea0d5df)

The result is this in the occurrence editor:
![image](https://github.com/BioKIC/Symbiota/assets/16688784/e08bcd12-d067-41c8-a83b-91ace48bfc85)

and this in the determination history:
![image](https://github.com/BioKIC/Symbiota/assets/16688784/d828762d-d2d1-4913-9790-067a65212346)

What we WANT is a determination history that looks like this:
![image](https://github.com/BioKIC/Symbiota/assets/16688784/36126c1d-8de5-4805-84f8-6eb76a18c1a4)

We have created a new determination based on the source record, if it's not identical to the sink record. "Unknown" and "s.d." can be used for identifiedBy and dateIdentified, respectively, if these fields are null.

themerekat reviewed (changes requested):
See comment on main thread

themerekat commented:
From meeting today: Logan is going to ask Ed whether we should just do the work that will be necessary to make sure that ALL records that have data in Latest Identification have at least one Determination. This would make error catching of the exception unnecessary.

MuchQuak committed:
updating local branch with dev

MuchQuak committed:
adds copies of some functions from occurrenceEditorDeterminations to properly merge determinations records

copied functions have been label as such with todos to remove in 3.2
when occurrence editor work to remove latest identification section has
been done

MuchQuak committed:
removing dead code comments

themerekat commented:
@MuchQuak , ack! The wrong determination is being prioritized when merging records. The Current determination should be that of the "good" record (i.e., the one that is listed in the pop up window). Right now, the determination from the record in the main data entry screen is being prioritized instead.

themerekat reviewed (changes requested):
See latest comment

MuchQuak committed:
fix determination merging priority to favor the lasting record

MuchQuak committed:
adds temp function to transfer attributes over to omoccuridentifers if none are present before record merge takes place

themerekat reviewed (approved):
It's working!!
